internal/security/redis/credentialset.go

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright 2020 Redis Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Andre Srinivasan
+ *******************************************************************************/
+
+package redis
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	dbContainer "github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/container"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db/redis"
+	"github.com/edgexfoundry/edgex-go/internal/security/redis/container"
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/startup"
+	bootstrapConfig "github.com/edgexfoundry/go-mod-bootstrap/config"
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+)
+
+func credentialSet(ctx context.Context, _ *sync.WaitGroup, startupTimer startup.Timer, dic *di.Container) bool {
+	var err error
+
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+
+	var credentials bootstrapConfig.Credentials = bootstrapConfig.Credentials{
+		Username: "unset",
+		Password: "unset",
+	}
+	for startupTimer.HasNotElapsed() {
+		credentials, err = bootstrapContainer.CredentialsProviderFrom(dic.Get).GetDatabaseCredentials(container.ConfigurationFrom(dic.Get).Databases["Primary"])
+		if err == nil {
+			break
+		}
+		lc.Warn(fmt.Sprintf("Could not retrieve database credentials (startup timer has not expired): %v", err.Error()))
+		startupTimer.SleepForInterval()
+	}
+
+	if credentials.Password == "unset" {
+		lc.Error("Failed to retrieve database credentials before startup timer expired")
+		return false
+	}
+
+	redisConn := dbContainer.DBClientFrom(dic.Get).(*redis.Client).Pool.Get()
+	_, err = redisConn.Do("CONFIG", "SET", "REQUIREPASS", credentials.Password)
+	if err != nil {
+		lc.Error("Could not set Redis password: %v", err.Error())
```

Comment:
When I run this I see %v in the output.  Did you mean %s instead?

Reply:
Corrected